x-pack/solutions/observability/plugins/apm/server/routes/alerts/rule_types/utils/unflatten_object.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { set } from '@kbn/safer-lodash-set';
+
+export interface GroupByFields {
+  [x: string]: any;
+}
+
+export const unflattenObject = <T extends object = GroupByFields>(object: object): T =>
```

Comment:
Shall we have a test for this?

Reply:
added test in acdddd148ce1a551a32caeb6f6b178b31ace08f5